include/LadspaBase.h

```diff
@@ -35,30 +35,20 @@ namespace lmms
 
 class LadspaControl;
 
-
-typedef enum BufferRates
-{
+using buffer_rate_t = enum BufferRates {
 	CHANNEL_IN,
 	CHANNEL_OUT,
 	AUDIO_RATE_INPUT,
 	AUDIO_RATE_OUTPUT,
 	CONTROL_RATE_INPUT,
 	CONTROL_RATE_OUTPUT
-} buffer_rate_t;
+};
 
-typedef enum BufferData
-{
-	TOGGLED,
-	ENUM,
-	INTEGER,
-	FLOATING,
-	TIME,
-	NONE
-} buffer_data_t;
+using buffer_data_t = enum BufferData { TOGGLED, ENUM, INTEGER, FLOATING, TIME, NONE };
 
 //! This struct is used to hold port descriptions internally
 //! which where received from the ladspa plugin
-typedef struct PortDescription
+using port_desc_t = struct PortDescription
```

Comment:
Removing ``PortDescription`` gives back errors in LadspaControl.h and LadspaControl.cpp. The GCC compiler complains about ``port_desc_t`` being an incomplete type. I also think ``PortDescription`` is used in some parts of the code, rather than just ``port_desc_t``. Removing ``struct PortDescription`` would probably mean more "out-of-focus" changes in terms of this PR, if that makes sense. For simplicity's sake, I am going to leave it as is.

Reply:
Because in `typedef struct PortDescription port_desc_t;` in `LadspaControl.h`, `struct PortDescription` is [implicitly forward-declared](https://en.cppreference.com/w/cpp/language/elaborated_type_specifier).

Reply:
Yeah, I think this means we can't get around the `struct` keyword here then, because it's a forward declaration meaning `port_desc_t` is a struct (Same as you write `class QLabel;` to forward declare `QLabel` as a class). Is that correct?

Reply:
I think you can remove `PortDescription` and replace `typedef struct PortDescription port_desc_t;` with `struct port_desc_t;`.